traitsui/editor.py

```diff
@@ -406,6 +420,58 @@ def parse_extended_name(self, name):
 
         return (object, name, partial(xgetattr, object, name))
 
+    def set_tooltip(self, control=None):
+        """ Sets the tooltip for a specified toolkit control.
+
+        This uses the tooltip_text method to get the text to use.
+
+        Parameters
+        ----------
+        control : optional toolkit control
+            The toolkit control that is having the tooltip set.  If None
+            then the editor's control is used.
+
+        Returns
+        -------
+        tooltip_set : bool
+            Whether or not a tooltip value could be set.
+        """
+        text = self.tooltip_text()
+        if text is None:
+            return False
+
+        if control is None:
+            control = self.control
+
+        self.set_tooltip_text(control, text)
+
+        return True
+
+    def tooltip_text(self):
+        """ Get the text for a tooltip, checking various sources.
+
+        This checks for text from, in order:
+
+        - the editor's description trait
+        - the base trait's 'tooltip' metadata
+        - the base trait's 'desc' metadata
+
+        Returns
+        -------
+        text : str or None
+            The text for the tooltip, or None if no suitable text can
+            be found.
+        """
+        text = self.description
+        if not text:
+            trait = self.object.base_trait(self.name)
+            text = trait.tooltip
+            if text is None:
+                text = trait.desc
+                if text is not None:
+                    text = "Specifies " + text
+        return text
```

Comment:
Yep, that's better (I think I'd shift the desc_metadata later, but otherwise it's good).  I did it like the above as it was copy-pasting the original code with some slight modifications, but keeping the logic as-is.

If you like I can make the change in this PR, or we could have a separate PR to make the change clearer from an historical context.